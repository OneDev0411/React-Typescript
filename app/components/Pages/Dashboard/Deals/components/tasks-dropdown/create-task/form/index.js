import React from 'react'
import {
  ChecklistNewItemInputContainer,
  ChecklistNewItemInput,
  ChecklistNewItemCancel,
  ChecklistItemContainer,
  ChecklistItemNotifyOffice,
  ChecklistItemTitle
} from '../../styled'
import { NotifyOffice } from '../../notify-office'

export class CreateTaskForm extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      notifyOffice: !props.checklist.is_deactivated
    }
  }

  toggleNotifyOffice = () =>
    this.setState({ notifyOffice: !this.state.notifyOffice })

  onKeyPress = e => {
    const { onFinish, checklist } = this.props
    const { notifyOffice } = this.state
    const { value } = e.target

    if (value.length === 0) {
      return false
    }

    if (e.which === 13) {
      onFinish(checklist.id, value, notifyOffice)
    }
  }

  render() {
    const { notifyOffice } = this.state
    const { checklist, onCancel, isSaving } = this.props

    if (isSaving) {
      return (
        <ChecklistItemContainer onClick={e => e.stopPropagation()}>
          <ChecklistItemTitle color="#2196f3" bold>
            Creating task ...
          </ChecklistItemTitle>
        </ChecklistItemContainer>
      )
    }

    return (
      <ChecklistItemContainer onClick={e => e.stopPropagation()}>
        <ChecklistItemTitle>
          <ChecklistNewItemInputContainer>
            <ChecklistNewItemInput
              autoFocus
              placeholder="Name task and press enter"
              onKeyPress={this.onKeyPress}
            />
            <ChecklistNewItemCancel onClick={onCancel}>
              Cancel
            </ChecklistNewItemCancel>
          </ChecklistNewItemInputContainer>
        </ChecklistItemTitle>

        <ChecklistItemNotifyOffice isSelected>
          <NotifyOffice
            checklist={checklist}
            isSelected={notifyOffice}
            onChange={this.toggleNotifyOffice}
          />
        </ChecklistItemNotifyOffice>
      </ChecklistItemContainer>
    )
  }
}
