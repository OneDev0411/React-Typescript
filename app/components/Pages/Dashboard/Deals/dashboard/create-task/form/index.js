import React from 'react'
import { connect } from 'react-redux'
import { DropdownButton, MenuItem } from 'react-bootstrap'
import _ from 'underscore'
import { createTask } from '../../../../../../../store_actions/deals'

const createNewTask = (form, props) => {
  const { dealId, createTask, listId } = props

  createTask(dealId, form.id, form.name, 'Incomplete', 'Form', listId)
}

const CreateForm = (props) => {
  const { forms } = props

  return (
    <div className="creator task-form">
      <DropdownButton
        noCaret
        className="add-task"
        id="dropdown-create-form"
        title="+ Add New"
      >
        {
          _.map(forms, form => (
            <MenuItem
              key={`FORM_ITEM_${form.id}`}
              onClick={() => createNewTask(form, props)}
            >
              { form.name }
            </MenuItem>
          ))
        }
      </DropdownButton>
    </div>
  )
}

export default connect(({ deals }) => ({
  forms: deals.forms
}), { createTask })(CreateForm)
