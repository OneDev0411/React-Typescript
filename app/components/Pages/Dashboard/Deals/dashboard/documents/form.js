import React from 'react'
import { connect } from 'react-redux'
import { Row, Col } from 'react-bootstrap'
import { editForm } from '../../../../../../store_actions/deals/forms'
import ESignButton from '../esign/button'

const Form = ({
  task,
  editForm
}) => {
  if (!task || !task.form) {
    return false
  }

  const { submission } = task
  const attachments = submission && submission.state === 'Fair' ? [task.id] : []

  return (
    <div className="file">
      <div className="title">Digital Form</div>
      <div className="file-group">
        <Row className="item digital-form">
          <Col sm={1} xs={12} className="image vcenter">
            <img src="/static/images/deals/digital-form.svg" />
          </Col>

          <Col sm={5} xs={12} className="name vcenter">
            <div>Digital Form</div>
          </Col>

          <Col sm={6} xs={12} className="actions vcenter">
            <ESignButton
              dealId={task.deal}
              task={task}
              attachments={attachments}
            />
            <button className="btn-deal" onClick={() => editForm(task)}>Edit</button>
          </Col>
        </Row>
      </div>
    </div>
  )
}

export default connect(null, { editForm })(Form)
