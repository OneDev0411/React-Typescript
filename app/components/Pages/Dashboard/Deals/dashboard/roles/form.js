import React from 'react'
import { FormGroup, FormControl, DropdownButton, MenuItem } from 'react-bootstrap'
import roleNames from '../../utils/roles'

const role_names = [
  // 'BuyerAgent',
  'CoBuyerAgent',
  // 'SellerAgent',
  'CoSellerAgent',
  'Buyer',
  'Seller',
  'Title',
  'Lawyer',
  'Lender',
  'TeamLead',
  'Appraiser',
  'Inspector',
  'Tenant',
  'Landlord'
]

export default class Form extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      validation: {},
      form: props.form || {}
    }
  }

  setForm(field, value) {
    const { form } = this.state

    this.setState({
      form: {
        ...form,
        [field]: value
      }
    }, this.validate)
  }

  validate() {
    const fields = ['first_name', 'last_name', 'email', 'role']
    const { form } = this.state
    const validation = {}
    let validated = true

    fields.forEach(key => {
      if (!form[key] || form[key].length === 0) {
        validation[key] = 'error'
        validated = false
      }
    })

    this.setState({ validation })
    this.props.onFormCompleted(validated ? form : null)
  }

  render() {
    const { allowedRoles } = this.props
    const { form, validation } = this.state

    return (
      <div>
        <input
          className="first_name"
          placeholder="First Name"
          value={form.first_name || ''}
          onChange={e => this.setForm('first_name', e.target.value)}
        />

        <input
          className="last_name"
          placeholder="Last Name"
          value={form.last_name || ''}
          onChange={e => this.setForm('last_name', e.target.value)}
        />

        <input
          className="email"
          placeholder="Email"
          value={form.email || ''}
          onChange={e => this.setForm('email', e.target.value)}
        />

        <input
          className="phone"
          placeholder="Phone"
          value={form.phone || ''}
          onChange={e => this.setForm('phone', e.target.value)}
        />

        <DropdownButton
          id="deal-add-role--drp"
          className="deal-add-role--drp"
          title={form.role || 'Select Role'}
        >
          {
            role_names
            .filter(name => {
              if (!allowedRoles) {
                return true
              }

              return allowedRoles.indexOf(name) > -1
            })
            .map(name =>
              <MenuItem
                key={`ROLE_${name}`}
                onClick={() => this.setForm('role', name)}
              >
                { roleNames(name) }
              </MenuItem>
            )
          }

        </DropdownButton>

        <div className="role-descr">
          Just select the role of the signer and Rechat already knows where they need to sign.
        </div>
      </div>
    )
  }
}
