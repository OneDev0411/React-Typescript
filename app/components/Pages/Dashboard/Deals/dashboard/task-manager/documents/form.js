import React from 'react'
import { connect } from 'react-redux'
import { Row, Col } from 'react-bootstrap'

export default ({
  task
}) => (
  <div className="deal-files-form">
    {
      task && task.form &&
      <Row className="file">
        <Col sm={1} xs={12} className="image vcenter">
          <img src="/static/images/deals/form.png" />
        </Col>

        <Col sm={5} xs={12} className="name vcenter">
          <div>Form</div>
          <div className="note">May 10 17 @ 1:15 AM</div>
        </Col>

        <Col sm={6} xs={12} className="actions vcenter">
          <button>eSigns</button>
          <button>Edit</button>
        </Col>
      </Row>
    }

  </div>
)
