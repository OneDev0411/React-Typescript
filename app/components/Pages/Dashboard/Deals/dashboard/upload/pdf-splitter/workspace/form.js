import React from 'react'
import { connect } from 'react-redux'
import { Button, ProgressBar } from 'react-bootstrap'
import cn from 'classnames'
import _ from 'underscore'
import { addNotification as notify } from 'reapop'
import TasksDropDown from '../../tasks-dropdown'
import Checkbox from '../../../../components/radio'
import Deal from '../../../../../../../../models/Deal'
import {
  resetSplitterSelectedPages,
  resetSplitter,
  changeNeedsAttention,
  addAttachment
} from '../../../../../../../../store_actions/deals'

class WorkspaceForm extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      saving: false,
      title: '',
      task: props.upload.task || null,
      notifyOffice: true
    }
  }

  isFormValidated() {
    const { title, task } = this.state
    const { splitter } = this.props

    if (title.length > 0 && task !== null && _.size(splitter.pages) > 0) {
      return true
    }

    return false
  }

  async save() {
    const { title, task, notifyOffice } = this.state
    const { notify, upload, splitter } = this.props
    const { pages } = splitter
    let created = false

    // set status
    this.setState({ saving: true })

    const files = _.chain(pages)
      .pluck('documentId')
      .uniq()
      .map(id => {
        const file = upload.files[id].fileObject
        file.id = id
        return file
      })
      .value()

    try {
      const { file } = await Deal.splitPDF(title, task.room.id, files, pages)

      // add files to attachments list
      this.props.addAttachment(task.deal, task.checklist, task.id, file)

      // set create as true
      created = true

      if (notifyOffice) {
        this.props.changeNeedsAttention(task.id, true)
      }

      notify({
        message: `Pdf "${title}" created successfully`,
        status: 'success'
      })

    } catch(e) {
      notify({
        message: "Couldn't create pdf file. try again.",
        status: 'error'
      })
    }

    // set status
    this.setState({
      saving: false,
      title: '',
      notifyOffice: true
    })

    return created
  }

  async saveAndQuit() {
    const saved = await this.save()

    if (!saved) {
      return false
    }

    // destruct splitter states
    this.props.resetSplitter()
  }

  async saveAndNew() {
    const saved = await this.save()

    if (!saved) {
      return false
    }

    // reset selected pages
    this.props.resetSplitterSelectedPages()
  }

  render() {
    const { upload, tasks } = this.props
    const { title, task, notifyOffice, saving } = this.state
    const formValidated = this.isFormValidated()

    if (saving) {
      return (
        <div className="splitter-saving">
          <div className="inner">
            Saving PDF... (It might take a few moments)
            <ProgressBar now={100} bsStyle="success" active />
          </div>
        </div>
      )
    }

    return (
      <div className="details">
        <input
          className="title"
          placeholder="Document title ..."
          value={title}
          onChange={e => this.setState({ title: e.target.value })}
        />

        <TasksDropDown
          onSelectTask={(taskId) => this.setState({ task: tasks[taskId] })}
          selectedTask={task}
          upload={upload}
        />

        <Checkbox
          square
          selected={notifyOffice}
          title="Notify Office"
          onClick={() => this.setState({ notifyOffice: !notifyOffice })}
        />

        <div className="buttons">
          <button
            onClick={() => this.saveAndQuit()}
            disabled={!formValidated}
            className={cn('save-quit', { disabled: !formValidated })}
          >
            Save and quit
          </button>

          <button
            className={cn('save-new', { disabled: !formValidated })}
            disabled={!formValidated}
            onClick={() => this.saveAndNew()}
          >
            Save and create another
          </button>
        </div>

      </div>
    )
  }
}

function mapStateToProps({ deals }) {
  return {
    upload: deals.upload,
    splitter: deals.splitter,
    tasks: deals.tasks
  }
}

export default connect(mapStateToProps, {
  notify,
  resetSplitter,
  resetSplitterSelectedPages,
  changeNeedsAttention,
  addAttachment
})(WorkspaceForm)
