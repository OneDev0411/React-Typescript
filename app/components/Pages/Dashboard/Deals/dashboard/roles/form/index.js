import React, { Fragment } from 'react'
import { Form } from 'react-final-form'
import _ from 'underscore'
import { FormContainer } from './form-container'
import { Modal } from 'react-bootstrap'
import { ROLE_NAMES } from '../../../utils/roles'
import ActionButton from '../../../../../../../views/components/Button/ActionButton'
import CancelButton from '../../../../../../../views/components/Button/CancelButton'

export class RoleFormModal extends React.Component {
  getInitialValues = () => {
    const { form, isSubmitting } = this.props

    if (isSubmitting) {
      return this.formObject
    }

    this.formObject = {
      ...form,
      ...this.preselectRole,
      ...this.commissionAttributes
    }

    return this.formObject
  }

  /**
   * preselect role, if there is only one allowed role to select
   */
  get preselectRole() {
    const { form } = this.props
    const formRole = form && form.role

    if (formRole) {
      return {}
    }

    const availableRoles = ROLE_NAMES.filter(name => this.isAllowedRole(name))
    const preselectedRole = availableRoles.length === 1 && availableRoles[0]

    if (!preselectedRole) {
      return {}
    }

    return {
      role: preselectedRole
    }
  }

  /**
   * returns commission attributes
   */
  get commissionAttributes() {
    const { form } = this.props

    if (form && form.commission_percentage) {
      return {
        commission: form.commission_percentage,
        commission_type: 'commission_percentage'
      }
    }

    if (form && form.commission_dollar) {
      return {
        commission: form.commission_dollar,
        commission_type: 'commission_dollar'
      }
    }

    return {
      commission: '',
      commission_type: 'commission_percentage'
    }
  }

  onSubmit = values => {
    // keeps the last object of submitted form
    this.formObject = values

    // send form to the parent
    this.props.onFormSubmit(this.normalizeForm(values))
  }

  validate = async values => {
    const errors = {}
    const requiredFields = this.getRequiredFields(values)
    const validators = this.getFormValidators(requiredFields)

    requiredFields.forEach(fieldName => {
      if (!values[fieldName]) {
        errors[fieldName] = 'Required'
      }
    })

    await Promise.all(
      _.map(values, async (fieldValue, fieldName) => {
        const validator = validators[fieldName]

        if (!validator) {
          return false
        }

        if (!errors[fieldName] && !await validator(fieldValue)) {
          errors[fieldName] = this.errorNames[fieldName]
        }
      })
    )

    return errors
  }

  /**
   * get normalized form
   * commission logic: commission_type + commission = commission_<type>
   */
  normalizeForm = values => {
    const newValues = {}
    const { commission, commission_type } = values
    const ingoreFields = ['commission', 'commission_type', 'user']

    if (commission_type === 'commission_dollar') {
      newValues.commission_dollar = parseFloat(commission)
    } else if (commission_type === 'commission_percentage') {
      newValues.commission_percentage = parseFloat(commission)
    }

    return _.omit(
      {
        ...values,
        ...newValues
      },
      (fieldValue, fieldName) => ingoreFields.includes(fieldName)
    )
  }

  /**
   * get required fields based on different scenarios
   * @param {Object} values - couples of form {Field:Value}
   */
  getRequiredFields = values => {
    const list = ['role']

    const { company_title, role } = values

    if (
      company_title &&
      this.isValidString(company_title, [], 'company_title')
    ) {
      list.push('company_title')
    } else {
      list.push('legal_first_name', 'legal_last_name')
    }

    if (this.isEmailRequired(role)) {
      list.push('email')
    }

    if (this.isCommissionRequired(role)) {
      list.push('commission')
    }

    // when adding an agent, company should be mandatory
    if (role && role.includes('Agent')) {
      list.push('legal_first_name', 'legal_last_name', 'company_title')
    }

    return _.uniq(list)
  }

  /**
   * get form validators
   */
  getFormValidators = requiredFields => ({
    role: role => role,
    legal_prefix: value => this.isValidLegalPrefix(value, requiredFields),
    legal_last_name: name =>
      this.isValidString(name, requiredFields, 'legal_last_name'),
    legal_middle_name: name =>
      this.isValidString(name, requiredFields, 'legal_middle_name'),
    legal_first_name: name =>
      this.isValidString(name, requiredFields, 'legal_first_name'),
    company_title: name =>
      this.isValidString(name, requiredFields, 'company_title'),
    email: email => this.isValidEmail(email, requiredFields),
    phone_number: phoneNumber =>
      this.isValidPhoneNumber(phoneNumber, requiredFields),
    commission: value => this.isValidCommission(value, requiredFields)
  })

  /**
   * get error names
   */
  get errorNames() {
    return {
      email: 'Invalid Email Address',
      phone_number: 'Phone Number is invalid (###)###-####',
      commission: 'Invalid Commission value'
    }
  }

  /**
   * check role type is allowed to select or not
   */
  isAllowedRole = (name, formRole) => {
    const { deal, form, allowedRoles } = this.props

    const dealType = deal ? deal.deal_type : null

    if (
      ((name === 'BuyerAgent' || formRole === 'BuyerAgent') &&
        dealType === 'Buying') ||
      ((name === 'SellerAgent' || formRole === 'SellerAgent') &&
        dealType === 'Selling')
    ) {
      return false
    }

    if (!allowedRoles || (!this.isNewRecord && form.role === name)) {
      return true
    }

    return allowedRoles.includes(name)
  }

  /**
   * validate email
   */
  isValidEmail = (email, requiredFields) => {
    if (!email && !requiredFields.includes('email')) {
      return true
    }

    const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/

    return re.test(email)
  }

  /**
   * validate legal prefix
   */
  isValidLegalPrefix = (prefix, requiredFields) => {
    if (!prefix && !requiredFields.includes('legal_prefix')) {
      return true
    }

    return ['Mr', 'Mrs', 'Miss', 'Ms', 'Dr'].includes(prefix)
  }

  /**
   * validate phone number
   */
  isValidPhoneNumber = async (phoneNumber, requiredFields) => {
    if (!phoneNumber && !requiredFields.includes('phone_number')) {
      return true
    }

    const {
      PhoneNumberUtil
    } = await import('google-libphonenumber' /* webpackChunkName: "glpn" */)
    const phoneUtil = PhoneNumberUtil.getInstance()

    try {
      return phoneUtil.isValidNumber(phoneUtil.parse(phoneNumber, 'US'))
    } catch (e) {
      return false
    }
  }

  /**
   * validate commission value
   */
  isValidCommission = (commission, requiredFields) => {
    if (!commission && !requiredFields.includes('commission')) {
      return true
    }

    if (/[0-9]{1,10}(\.[0-9]{1,2})?/.test(commission)) {
      return true
    }

    return false
  }

  /**
   * validate string value
   */
  isValidString = (str, requiredFields, fieldName) => {
    if (!str && !requiredFields.includes(fieldName)) {
      return true
    }

    return str && str.trim().length > 0
  }

  /**
   * check email is required or not
   * https://gitlab.com/rechat/web/issues/563
   */
  isEmailRequired = role => ['BuyerAgent', 'SellerAgent'].includes(role)

  /**
   * check whether commission is required or not
   * it's required by default
   * https://gitlab.com/rechat/web/issues/691
   */
  isCommissionRequired = role => {
    const { deal, isCommissionRequired } = this.props

    // https://gitlab.com/rechat/web/issues/760
    if (deal && deal.deal_type === 'Buying' && role === 'SellerAgent') {
      return false
    }

    return this.shouldShowCommission(role) && isCommissionRequired !== false
  }

  /**
   * check whether should show commission or not
   */
  shouldShowCommission = role =>
    [
      'CoBuyerAgent',
      'BuyerAgent',
      'BuyerReferral',
      'CoSellerAgent',
      'SellerAgent',
      'SellerReferral'
    ].includes(role)

  /**
   * returns Submit button's caption
   */
  get submitCaption() {
    const { isSubmitting } = this.props

    if (isSubmitting) {
      return this.isNewRecord ? 'Saving...' : 'Updating...'
    }

    return this.isNewRecord ? 'Add' : 'Update'
  }

  /**
   * get form is new record or not
   */
  get isNewRecord() {
    const { form } = this.props

    return !form || !form.role
  }

  render() {
    const { form, isOpen, modalTitle, onHide, isSubmitting } = this.props

    return (
      <Modal
        dialogClassName="deals__role-form--modal"
        show={isOpen}
        onHide={onHide}
        backdrop="static"
      >
        <Modal.Header closeButton>{modalTitle}</Modal.Header>
        <Form
          onSubmit={this.onSubmit}
          validate={this.validate}
          initialValues={this.getInitialValues()}
          render={({ handleSubmit, values, pristine, invalid }) => (
            <Fragment>
              <Modal.Body
                className="u-scrollbar--thinner"
                style={{ padding: 0 }}
              >
                <FormContainer
                  form={form}
                  values={values}
                  handleSubmit={handleSubmit}
                  shouldShowCommission={this.shouldShowCommission}
                  isAllowedRole={this.isAllowedRole}
                  requiredFields={this.getRequiredFields(values)}
                />
              </Modal.Body>
              <Modal.Footer>
                <CancelButton disabled={isSubmitting} onClick={onHide}>
                  Canecl
                </CancelButton>
                <ActionButton
                  onClick={() => handleSubmit(this.onSubmit)}
                  type="submit"
                  disabled={isSubmitting || pristine || invalid}
                >
                  {this.submitCaption}
                </ActionButton>
              </Modal.Footer>
            </Fragment>
          )}
        />
      </Modal>
    )
  }
}

export default RoleFormModal
