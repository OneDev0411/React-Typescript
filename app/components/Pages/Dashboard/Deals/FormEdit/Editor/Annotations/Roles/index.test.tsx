import React from 'react'
import { render } from 'enzyme'
import toJson from 'enzyme-to-json'

// eslint-disable-next-line import/no-unresolved
import deal from 'fixtures/deal/live-seller'

// eslint-disable-next-line import/no-unresolved
import form from 'fixtures/deal/forms/listing-agreement'

import { Roles } from '.'

describe('Deal Form: Roles Container', () => {
  it('renders the component', () => {
    const wrapper = render(
      <Roles
        deal={deal}
        roles={deal.roles}
        pageIndex={form.pageIndex}
        annotations={form.annotations}
        values={form.values}
      />
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
