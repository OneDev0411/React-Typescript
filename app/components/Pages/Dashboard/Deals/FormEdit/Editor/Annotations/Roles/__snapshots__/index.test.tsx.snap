// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deal Form: Roles Container renders the component 1`] = `
Array [
  <div
    style="position:absolute;left:132.3529411764706px;top:278.11764705882354px;width:714px;height:16px;font-size:14px;font-family:Arial;color:rgb(0, 0, 128);font-weight:normal;min-width:15px;padding:0 3px;cursor:pointer;background-color:#d2e5f2"
  >
    <div
      style="display:inline-block"
    />
    <span
      class="styled__Container-sc-1fyjv7w-0 hAsrBp"
    >
      <svg
        class="IconAddCircleOutline__Image-fo18hq-0 hSkVmB"
        style="height:16px;fill:rgb(0, 0, 128)"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
        />
      </svg>
    </span>
  </div>,
  <div
    style="position:absolute;left:132.3529411764706px;top:296.6470588235294px;width:714px;height:16px;font-size:14px;font-family:Arial;color:rgb(0, 0, 128);font-weight:normal;min-width:15px;padding:0 3px;cursor:pointer;background-color:#d2e5f2"
  >
    <div
      style="display:inline-block"
    />
    <span
      class="styled__Container-sc-1fyjv7w-0 hAsrBp"
    >
      <svg
        class="IconAddCircleOutline__Image-fo18hq-0 hSkVmB"
        style="height:16px;fill:rgb(0, 0, 128)"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
        />
      </svg>
    </span>
  </div>,
  <div
    style="position:absolute;left:174.1764705882353px;top:315.3529411764706px;width:673px;height:16px;font-size:14px;font-family:Arial;color:rgb(0, 0, 128);font-weight:normal;min-width:15px;padding:0 3px;cursor:pointer;background-color:transparent"
  >
    <div
      style="display:inline-block"
    >
      <span
        class="styled__RoleItem-bgrt01-0 bdePoE"
        style="font-size:14px"
      />
      <svg
        class="EditIcon__Image-sc-142t31x-0 GTkYE"
        style="height:16px;fill:rgb(0, 0, 128)"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04a.996.996 0 0 0 0-1.41l-2.34-2.34a.996.996 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </div>
  </div>,
  <div
    style="position:absolute;left:220.94117647058826px;top:333.88235294117635px;width:626px;height:16px;font-size:14px;font-family:Arial;color:rgb(0, 0, 128);font-weight:normal;min-width:15px;padding:0 3px;cursor:pointer;background-color:transparent"
  >
    <div
      style="display:inline-block"
    >
      <span
        class="styled__RoleItem-bgrt01-0 bdePoE"
        style="font-size:14px"
      />
      <svg
        class="EditIcon__Image-sc-142t31x-0 GTkYE"
        style="height:16px;fill:rgb(0, 0, 128)"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04a.996.996 0 0 0 0-1.41l-2.34-2.34a.996.996 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </div>
  </div>,
  <div
    style="position:absolute;left:157.23529411764707px;top:352.58823529411757px;width:372px;height:16px;font-size:14px;font-family:Arial;color:rgb(0, 0, 128);font-weight:normal;min-width:15px;padding:0 3px;cursor:pointer;background-color:#d2e5f2"
  >
    <div
      style="display:inline-block"
    />
    <span
      class="styled__Container-sc-1fyjv7w-0 hAsrBp"
    >
      <svg
        class="IconAddCircleOutline__Image-fo18hq-0 hSkVmB"
        style="height:16px;fill:rgb(0, 0, 128)"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
        />
      </svg>
    </span>
  </div>,
  <div
    style="position:absolute;left:160.76470588235296px;top:371.11764705882354px;width:686px;height:16px;font-size:14px;font-family:Arial;color:rgb(0, 0, 128);font-weight:normal;min-width:15px;padding:0 3px;cursor:pointer;background-color:#d2e5f2"
  >
    <div
      style="display:inline-block"
    />
    <span
      class="styled__Container-sc-1fyjv7w-0 hAsrBp"
    >
      <svg
        class="IconAddCircleOutline__Image-fo18hq-0 hSkVmB"
        style="height:16px;fill:rgb(0, 0, 128)"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
        />
      </svg>
    </span>
  </div>,
  <div
    style="position:absolute;left:136.05882352941177px;top:405px;width:711px;height:16px;font-size:14px;font-family:Arial;color:rgb(0, 0, 128);font-weight:normal;min-width:15px;padding:0 3px;cursor:pointer;background-color:transparent"
  >
    <div
      style="display:inline-block"
    >
      <span
        class="styled__RoleItem-bgrt01-0 bdePoE"
        style="font-size:14px"
      >
        Rechat
      </span>
    </div>
  </div>,
  <div
    style="position:absolute;left:174.1764705882353px;top:423.52941176470586px;width:673px;height:16px;font-size:14px;font-family:Arial;color:rgb(0, 0, 128);font-weight:normal;min-width:15px;padding:0 3px;cursor:pointer;background-color:transparent"
  >
    <div
      style="display:inline-block"
    >
      <span
        class="styled__RoleItem-bgrt01-0 bdePoE"
        style="font-size:14px"
      >
        Mckinney Avenue
      </span>
    </div>
  </div>,
  <div
    style="position:absolute;left:220.94117647058826px;top:442.2352941176471px;width:626px;height:16px;font-size:14px;font-family:Arial;color:rgb(0, 0, 128);font-weight:normal;min-width:15px;padding:0 3px;cursor:pointer;background-color:transparent"
  >
    <div
      style="display:inline-block"
    >
      <span
        class="styled__RoleItem-bgrt01-0 bdePoE"
        style="font-size:14px"
      >
        Dallas TX 75081
      </span>
    </div>
  </div>,
  <div
    style="position:absolute;left:157.23529411764707px;top:460.7647058823529px;width:372px;height:16px;font-size:14px;font-family:Arial;color:rgb(0, 0, 128);font-weight:normal;min-width:15px;padding:0 3px;cursor:pointer;background-color:transparent"
  >
    <div
      style="display:inline-block"
    >
      <span
        class="styled__RoleItem-bgrt01-0 bdePoE"
        style="font-size:14px"
      >
        (214) 999-9999
      </span>
    </div>
  </div>,
  <div
    style="position:absolute;left:565.4117647058824px;top:460.7647058823529px;width:281px;height:16px;font-size:14px;font-family:Arial;color:rgb(0, 0, 128);font-weight:normal;min-width:15px;padding:0 3px;cursor:pointer;background-color:transparent"
  >
    <div
      style="display:inline-block"
    >
      <span
        class="styled__RoleItem-bgrt01-0 bdePoE"
        style="font-size:14px"
      >
        (214) 570-0087
      </span>
    </div>
  </div>,
  <div
    style="position:absolute;left:160.76470588235296px;top:479.2941176470588px;width:686px;height:16px;font-size:14px;font-family:Arial;color:rgb(0, 0, 128);font-weight:normal;min-width:15px;padding:0 3px;cursor:pointer;background-color:transparent"
  >
    <div
      style="display:inline-block"
    >
      <span
        class="styled__RoleItem-bgrt01-0 bdePoE"
        style="font-size:14px"
      >
        trey+agent@rechat.com
      </span>
    </div>
  </div>,
  <div
    style="position:absolute;left:50.16808823529412px;top:1108.4383823529413px;width:163px;height:29px;font-size:5px;font-family:Arial;color:rgb(0, 0, 0);font-weight:normal;min-width:15px;padding:0;cursor:pointer;background-color:transparent;border:none"
  >
    Rechat
  </div>,
  <div
    style="position:absolute;left:212.50147058823532px;top:1108.4383823529413px;width:159px;height:29px;font-size:5px;font-family:Arial;color:rgb(0, 0, 0);font-weight:normal;min-width:15px;padding:0;cursor:pointer;background-color:transparent;border:none"
  >
    Mckinney Avenue
  </div>,
  <div
    style="position:absolute;left:408.2779411764706px;top:1117.2801470588236px;width:135px;height:15px;font-size:5px;font-family:Arial;color:rgb(0, 0, 0);font-weight:normal;min-width:15px;padding:0;cursor:pointer;background-color:transparent;border:none"
  >
    (214) 999-9999
  </div>,
  <div
    style="position:absolute;left:50.16808823529412px;top:1137.139705882353px;width:133px;height:14px;font-size:5px;font-family:Arial;color:rgb(0, 0, 0);font-weight:normal;min-width:15px;padding:0;cursor:pointer;background-color:transparent;border:none"
  >
    Trey Jones
  </div>,
]
`;
